be/src/vec/exec/scan/file_scanner.cpp

```diff
@@ -1480,6 +1489,21 @@ Status FileScanner::_generate_parititon_columns() {
     return Status::OK();
 }
 
+Status FileScanner::_generate_data_lake_partition_columns() {
+    _data_lake_partition_col_descs.clear();
+    if (_current_range.__isset.data_lake_partition_values) {
+        const auto& partition_values = _current_range.data_lake_partition_values;
+        for (const auto& [key, value] : partition_values) {
+            if (_all_col_name_to_slot_desc.find(key) == _all_col_name_to_slot_desc.end()) {
```

Comment:
`_all_col_name_to_slot_desc.find(key)` find the key, and `_all_col_name_to_slot_desc[key]` find again, should optimize to find only once